src/ideal/ideal.jl

```diff
@@ -1278,6 +1278,36 @@ function hilbert_series(I::sideal{spoly{T}}, w::Vector{<:Integer}) where T <: Ne
    return z
 end
 
+@doc raw"""
+    hilbert_series(I::sideal{spoly{T}}, Qt::PolyRing) where T <: Nemo.FieldElem
+
+Return the polynomial $Q(t)$ as element of Qt where `Q(t)/(1-t)^nvars(base_ring(I))`
+is the Hilbert-Poincare series of $I$ for weights $(1, \dots, 1)$.
+"""
+function hilbert_series(I::sideal{spoly{T}}, Qt::PolyRing) where T <: Nemo.FieldElem
+   I.isGB || error("Not a Groebner basis")
```

Comment:
If it is the caller's responsibility to ensure that GB for `I` is available then this should be in the doc!
Maybe consider an optional/kw arg which the user can use to permit `hilbert_series` to compute a GB if necessary?